src/proxy/http/metrics/report.rs

```diff
@@ -126,3 +142,49 @@ where
         Ok(())
     }
 }
+
+// === impl Scope ===
+
+impl Default for Scope {
+    fn default() -> Self {
+        Self {
+            request_total_key: "request_total".to_owned(),
+            response_total_key: "response_total".to_owned(),
+            response_latency_ms_key: "response_latency_ms".to_owned(),
+        }
+    }
+}
+
+impl Scope {
+    fn prefixed(prefix: &'static str) -> Self {
+        if prefix.is_empty() {
+            return Self::default();
+        }
+
+        Self {
+            request_total_key: format!("{}_request_total", prefix),
+            response_total_key: format!("{}_response_total", prefix),
+            response_latency_ms_key: format!("{}_response_latency_ms", prefix),
```

Comment:
This seems like it'll result in a lot of duplication of the strings "_request_total", "_response_total", and "_response_latency_ms"... if metrics memory usage is an issue, it seems like it should be possible to have this struct only store the prefix and interpolate it with those strings when we format the scope key? 

This is probably only worth bothering with if it turns out to be an issue, but I figured I'd ask about it.

Reply:
> This seems like it'll result in a lot of duplication of the strings

Why? Do you anticipate there being many instances of these?

Reply:
If this type is constructed on a per-route basis, it seems like there's the potential for that, in the case of a proxy which talks to a lot of services that each have a lot of routes...but the scale may never be great enough for it to matter.

Reply:
This is the _report_ type, which means that this is only constructed... once ;)

https://github.com/linkerd/linkerd2-proxy/blob/4e0a1f0100d1a87e7658e8ffe6906290c370f8f5/src/app/main.rs#L176-L179